src/selectors/filtered-pipeline.js

```diff
@@ -0,0 +1,116 @@
+ * @returns {Array} An array of edge objects.
+ */
+const getEdges = createSelector(
+  [getEdgeIDs, getEdgeSources, getEdgeTargets],
+  (edgeIDs, edgeSources, edgeTargets) =>
+    edgeIDs.map((id) => ({
+      id,
+      source: edgeSources[id],
+      target: edgeTargets[id],
+    }))
+);
+
+/**
+ * Selector to organize edges by their source and target nodes.
+ * @param {Array} edges - Array of edge objects.
+ * @returns {Object} An object containing edges mapped by source and target nodes.
+ */
+
+export const getEdgesByNode = createSelector([getEdges], (edges) => {
+  const sourceEdges = {};
+  const targetEdges = {};
+
+  for (const edge of edges) {
+    if (!sourceEdges[edge.target]) {
+      sourceEdges[edge.target] = [];
+    }
+
+    sourceEdges[edge.target].push(edge.source);
+
+    if (!targetEdges[edge.source]) {
+      targetEdges[edge.source] = [];
+    }
+
+    targetEdges[edge.source].push(edge.target);
+  }
+
+  return { sourceEdges, targetEdges };
+});
+
+/**
+ * Recursive function to find all linked nodes starting from a given node ID.
+ * @param {string} nodeID - The starting node ID.
+ * @param {Object} edgesByNode - A map of node IDs to their connected node IDs.
+ * @param {Object} visited - A map to keep track of visited nodes.
+ * @returns {Object} A map of visited nodes.
+ */
+
+const findLinkedNodes = (nodeID, edgesByNode, visited) => {
+  if (!visited[nodeID]) {
+    visited[nodeID] = true;
+    if (edgesByNode[nodeID]) {
+      edgesByNode[nodeID].forEach((nodeID) =>
+        findLinkedNodes(nodeID, edgesByNode, visited)
+      );
+    }
+  }
+
+  return visited;
+};
+
+const findNodesInBetween = (sourceEdges, startID, endID) => {
+  if (!startID || !endID) {
+    return [startID, endID].filter(Boolean);
+  }
+
+  const linkedNodesBeforeEnd = {};
+  findLinkedNodes(endID, sourceEdges, linkedNodesBeforeEnd);
+
+  const linkedNodeBeforeStart = {};
+  findLinkedNodes(startID, sourceEdges, linkedNodeBeforeStart);
+
+  let filteredNodeIDs = Object.keys(linkedNodesBeforeEnd);
+
+  if (filteredNodeIDs.includes(startID) && filteredNodeIDs.includes(endID)) {
+    // Remove any nodes before startID not including startID itself
+    Object.keys(linkedNodeBeforeStart).forEach((node) => {
+      if (node !== startID) {
+        const index = filteredNodeIDs.indexOf(node);
+        if (index > -1) {
+          filteredNodeIDs.splice(index, 1);
+        }
+      }
+    });
+  } else {
+    // If startID and endID are not connected, return only them
+    filteredNodeIDs = [startID, endID];
+  }
+
+  return filteredNodeIDs;
+};
+
+/**
+ * Selector to filter nodes that are connected between two specified node IDs.
+ * @param {Object} edgesByNode - Edges organized by node IDs.
+ * @param {string} startID - Starting node ID.
+ * @param {string} endID - Ending node ID.
+ * @returns {Array} Array of node IDs that are connected from startID to endID.
+ */
+
+export const getFilteredPipeline = createSelector(
```

Comment:
```suggestion
export const getSlicedPipeline = createSelector(
```

Reply:
I see we used filter terminology in all the places, if this is something the team feels fine to go ahead, then lets do that. For me this confuses with the nodeType, tags filter

Reply:
yeah i dont mind using sliced Pipeline. Let see what the team thinks in standup

Reply:
I also prefer using "slice" because it matches Kedro terminology and we have filters that behave differently. My original PR used "filtered" everywhere, which is probably why we followed that.

Reply:
cool happy to change to sliced Pipeline 👍

Reply:
DONE 👍 @rashidakanchwala @ravi-kumar-pilla